test/integration/logging_test.go

```diff
@@ -0,0 +1,166 @@
+		return nil
+	}, retry.Attempts(3))
+
+	if retryErr != nil {
+		t.Fatal(retryErr)
+	}
+
+	defer close(portForward)
+
+	_, errNamespace := clientset.CoreV1().Namespaces().Create(context.Background(), &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+		},
+	}, metav1.CreateOptions{})
+	if errNamespace != nil {
+		t.Fatal(errNamespace)
+	}
+
+	t.Cleanup(func() {
+		errNamespace = clientset.CoreV1().Namespaces().Delete(context.Background(), namespaceName, metav1.DeleteOptions{})
+		if errNamespace != nil {
+			t.Fatal(errNamespace)
+		}
+	})
+
+	_, roleBindingErr := clientset.RbacV1().RoleBindings(namespaceName).Create(context.Background(), &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "view-default-test",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      "default",
+				Namespace: namespaceName,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "view",
+		},
+	},
+		metav1.CreateOptions{})
+	if roleBindingErr != nil {
+		t.Fatal(roleBindingErr)
+	}
+
+	token, tokenErr := clientset.CoreV1().ServiceAccounts(namespaceName).CreateToken(context.Background(), "default", &authenticationv1.TokenRequest{}, metav1.CreateOptions{})
+	if tokenErr != nil {
+		fmt.Printf("could not create a token, %s", tokenErr)
+		t.Fatal(tokenErr)
+	}
+
+	clientHTTP := http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+
+	// Install the log-generator.
+	namespaceCmd := fmt.Sprintf("--namespace=%s", namespaceName)
+	cmd := exec.Command("bash", "../log-generators/cronjobs/install.sh", namespaceCmd, "--num-jobs=1")
+	output, errScript := cmd.CombinedOutput()
+	if errScript != nil {
+		fmt.Println("Could not run the log-generator: ", errScript)
+		t.Fatal(errScript)
+	}
+	fmt.Println("Output: ", string(output))
+
+	url := fmt.Sprintf("https://localhost:8081/apis/batch/v1/namespaces/%s/cronjobs/generate-log-1/log", namespaceName)
+	request, errRequest := http.NewRequest("GET", url, nil)
+	if errRequest != nil {
+		fmt.Printf("could not create a request, %s", errRequest)
+		t.Fatal(errRequest)
+	}
+	request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token.Status.Token))
+	fmt.Printf("HTTP request: %s\n", request.URL.RequestURI())
+
+	var bodyString string
+	retryErr = retry.Do(func() error {
+		response, errResp := clientHTTP.Do(request)
+		if errResp != nil {
+			fmt.Printf("Couldn't get a response HTTP, %s\n", errResp)
+			return errResp
+		}
+
+		body, errReadAll := io.ReadAll(response.Body)
+		if errReadAll != nil {
+			fmt.Printf("Couldn't read Body, %s\n", errReadAll)
+			return errReadAll
+		}
+
+		response.Body.Close()
+
+		if response.StatusCode != http.StatusOK {
+			fmt.Printf("HTTP body: %s\n", body)
+			fmt.Printf("The HTTP status returned is not OK, %s\n", response.Status)
+			return errors.New(response.Status)
+		}
```

Comment:
this could be replaced with a call to `test.GetUrl` if the part of parsing the body to a list of `Unstructured` instances is moved out.
See the other comment in the review

Reply:
Related to the abocve